extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java

```diff
@@ -76,14 +80,29 @@
     private long keyRefreshInterval;
 
     private ScheduledExecutorService executorService;
+
     @Inject
     private OkHttpClient okHttpClient;
 
-    private static Supplier<JWSSigner> createRsaPrivateKeySupplier(Oauth2Configuration configuration) {
+    private static Supplier<JWSSigner> createPrivateKeySupplier(Oauth2Configuration configuration) {
         return () -> {
             var pkId = configuration.getPrivateKeyAlias();
-            var pk = configuration.getPrivateKeyResolver().resolvePrivateKey(pkId, RSAPrivateKey.class);
-            return pk == null ? null : new RSASSASigner(pk);
+            var pk = configuration.getPrivateKeyResolver().resolvePrivateKey(pkId, PrivateKey.class);
+
+            JWSSigner signer = null;
+            if (pk != null && "EC".equals(pk.getAlgorithm())) {
```

Comment:
the null check is executed twice, maybe better to add a guard that returns null (or throws exception?) if pk is null and then check the algorithm.

Reply:
Done.